File: app/src/main/java/remix/myplayer/ui/fragment/base/BaseFragment.java
package remix.myplayer.ui.fragment.base;

import static remix.myplayer.ui.activity.base.BaseActivity.EXTERNAL_STORAGE_PERMISSIONS;

import android.content.Context;
import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.annotation.StringRes;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.Unbinder;
import remix.myplayer.App;
import remix.myplayer.util.Util;


/**
 * @ClassName
 * @Description
 * @Author Xiaoborui
 * @Date 2016/7/27 15:18
 */
public abstract class BaseFragment extends Fragment {

  protected Unbinder mUnBinder;
  protected Context mContext;
  protected boolean mHasPermission = false;
  protected String mPageName = BaseFragment.class.getSimpleName();

  @Override
  public void onAttach(Context context) {
    super.onAttach(context);
    mContext = context;
  }

  @Override
  public void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    mHasPermission = Util.hasPermissions(EXTERNAL_STORAGE_PERMISSIONS);
  }

  @Override
  public void onDestroyView() {
    super.onDestroyView();
    if (mUnBinder != null) {
      mUnBinder.unbind();
    }
  }

  public RecyclerView.Adapter getAdapter() {
    return null;
  }

  public void onResume() {
    super.onResume();
  }

  public void onPause() {
    super.onPause();
  }

  protected String getStringSafely(@StringRes int res) {
    if (isAdded()) {
      return getString(res);
    } else {
      return App.getContext().getString(res);
    }
  }

  protected String getStringSafely(@StringRes int res, Object... args) {
    if (isAdded()) {
      return getString(res, args);
    } else {
      return App.getContext().getString(res, args);
    }
  }

}

File: app/src/main/java/remix/myplayer/ui/fragment/base/BaseFragment.kt
package remix.myplayer.ui.fragment.base

import android.content.Context
import android.os.Bundle
import androidx.annotation.StringRes
import androidx.fragment.app.Fragment
import remix.myplayer.App
import remix.myplayer.util.PermissionUtil

/**
 * @ClassName
 * @Description
 * @Author Xiaoborui
 * @Date 2016/7/27 15:18
 */
abstract class BaseFragment : Fragment() {
  protected var hasPermission = false
  @JvmField
  protected var pageName = BaseFragment::class.java.simpleName

  override fun onAttach(context: Context) {
    super.onAttach(context)
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    hasPermission = PermissionUtil.hasNecessaryPermission()
  }

  override fun onDestroyView() {
    super.onDestroyView()
  }

  override fun onResume() {
    super.onResume()
  }

  override fun onPause() {
    super.onPause()
  }

  protected fun getStringSafely(@StringRes res: Int): String {
    return if (isAdded) {
      getString(res)
    } else {
      App.context.getString(res)
    }
  }

  protected fun getStringSafely(@StringRes res: Int, vararg args: Any?): String {
    return if (isAdded) {
      getString(res, *args)
    } else {
      App.context.getString(res, *args)
    }
  }
}